packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -552,3 +560,17 @@ defineProperty(LightningElement, 'CustomElementConstructor', {
 if (process.env.NODE_ENV !== 'production') {
     patchLightningElementPrototypeWithRestrictions(LightningElement.prototype);
 }
+
+defineProperty(LightningElement, 'shadow', {
+    get() {
+        return true;
+    },
+    set(v) {
```

Comment:
I would recommend adding a dev check here to make sure the `v` is actually a `boolean`.